components/script/dom/event.rs

```diff
@@ -433,6 +433,15 @@ impl EventMethods for Event {
         self.current_target.get()
     }
 
+    // https://dom.spec.whatwg.org/#dom-event-composedpath
+    fn ComposedPath(&self) -> Vec<DomRoot<EventTarget>> {
+        if let Some(target) = self.target.get() {
+            self.construct_event_path(&target)
```

Comment:
There is a TODO on that method for the shadow root stuff, it would be good to file an issue for it, and it would be a good follow-up if you are interested.

Reply:
I will create issue and follow up